crates/parser/src/grammar/functions.rs

```diff
@@ -205,10 +205,48 @@ pub fn parse_stmt(par: &mut Parser) -> ParseResult<Node<FuncStmt>> {
         Revert => parse_revert_stmt(par),
         Continue | Break | Pass => parse_single_word_stmt(par),
         Emit => parse_emit_statement(par),
+        Let => parse_var_decl(par),
         _ => parse_expr_stmt(par),
     }
 }
 
+fn parse_var_decl(par: &mut Parser) -> ParseResult<Node<FuncStmt>> {
+
+    let let_tkn = par.assert(TokenKind::Let);
+    let expr = parse_expr(par)?;
+    let target = expr_to_vardecl_target(par, expr.clone())?;
+    let node = match par.peek() {
+        Some(TokenKind::Colon) => {
+            par.next()?;
+            let typ = parse_type_desc(par)?;
+            let value = if par.peek() == Some(TokenKind::Eq) {
+                par.next()?;
+                Some(parse_expr(par)?)
+            } else {
+                None
+            };
+            let span = let_tkn.span + target.span + typ.span + value.as_ref();
+            par.expect_newline("variable declaration")?;
+            Node::new(FuncStmt::VarDecl { target, typ, value }, span)
+        },
+        _ => {
+            par.fancy_error(
+                "failed to parse variable declaration",
+                vec![Label::primary(
+                    expr.span,
+                    "Must be followed by type annotation",
+                )],
+                vec![
+                    "The left side of a variable declaration can be either a name\nor a non-empty tuple."
```

Comment:
This "left side of a variable ..." note doesn't really describe the error here (missing colon); could you remove it or replace it with something like "Example: `let x: u8 = 1`"?